server/routes/activities/daily-attendance-summary/handlers/notAttendedSelectDate.ts

```diff
@@ -18,12 +16,15 @@ export class NotAttendedDate {
   datePresetOption: DateOptions
 
   @ValidateIf(o => o.datePresetOption === DateOptions.OTHER)
-  @Type(() => SimpleDate)
-  @ValidateNested()
-  @IsValidDate({ message: 'Enter a valid date' })
-  @DateIsSameOrAfter(() => subDays(new Date(), 14), { message: 'Enter a date within the last 14 days' })
-  @DateIsSameOrBefore(() => addDays(new Date(), 60), { message: 'Enter a date up to 60 days in the future' })
-  date?: SimpleDate
+  @Transform(({ value }) => parseDatePickerDate(value))
+  @IsDate({ message: 'Enter a valid date' })
+  @DateValidator(thisDate => thisDate >= subDays(startOfToday(), 14), {
+    message: 'Enter a date within the last 14 days',
+  })
+  @DateValidator(thisDate => thisDate <= addDays(startOfToday(), 60), {
+    message: 'Enter a date up to 60 days in the future',
+  })
+  date?: Date
```

Comment:
I think we should move away from having multiple custom date validators and use a single generic date validator which accepts an evaluation method. This should allow us to remove all of our custom date validators.

Also, first transforming the date picker string to a date value allows us to use `class-validator`'s built-in `IsDate()` validator rather than using our custom `IsValidDate()` and `IsValidDatePickerDate()` validators.

Reply:
Nice change